babel/messages/catalog.py

```diff
@@ -35,7 +35,7 @@
             (?:\.(?:\*|[\d]+))?
             [hlL]?
         )
-        ([diouxXeEfFgGcrs%])
+        ((?<!\s)[diouxXeEfFgGcrs%])(?=([\s\'\)\.\,\:\"\!\]\>\?]|$))
```

Comment:
The commit message(s) don't seem to explain why these changes are necessary.

Reply:
Please read the original post at 

[Message.locations duplicate unnecessary #10104](https://github.com/sphinx-doc/sphinx/issues/10104)

This is an extract of the original message:

`By the way, in the past I posted a bug report mentioning the PYTHON_FORMAT problem, in that this re.Pattern causing the problem in recognizing this part "%50 'one letter'" (diouxXeEfFgGcrs%) as an ACCEPTABLE pattern, thus causing the flag "python_format" in the Message class to set, and the Catalog.dump_po will insert a "#, python-format" in the comment section of the message, causing applications such as PoEdit to flag up as a WRONG format for "python-format". The trick is to insert a look behind clause in the PYTHON_FORMAT pattern, as an example here:`

Reply:
After all the above comments, and as you finished your preview, could you please give me a approval as what other changes I should make, before I make another push to the pull request branch. Thank you.